Loader cannot handle components with required (no-default) fields

## Description

The scene/prefab loader cannot load components that have required fields without default values. It fails with "missing struct field" error.

## Steps to Reproduce

1. Use a component with a required field (no default):
```zig
// From labelle-tasks
pub const TaskItem = struct {
    item_type: ItemType,  // No default value
};
```

2. Try to use it in a .zon scene:
```zig
.{
    .components = .{
        .Sprite = .{ .name = "water.png" },
        .Task_item = .{ .item_type = .water },
    },
}
```

3. Build fails with:
```
loader.zig:429:45: error: missing struct field: item_type
            var component: ComponentType = .{};
```

## Root Cause

The loader initializes components with an empty struct literal `.{}` before applying field values:

```zig
var component: ComponentType = .{};  // Fails if any field has no default
```

## Expected Behavior

The loader should be able to construct components by applying all provided fields directly, without requiring defaults.

## Workaround

Create a wrapper component with default values:
```zig
pub const Task_item = struct {
    item_type: items.ItemType = .raw_food,  // Add default
};
```

But this means losing type safety for required fields and diverging from the upstream component definition.

## Suggested Fix

Use Zig's inline field initialization or a different construction pattern that doesn't require defaults for all fields.

🤖 Generated with [Claude Code](https://claude.ai/code)

--- a/src/loader.zig
+++ b/src/loader.zig
@@ -426,11 +426,9 @@ pub fn SceneLoader(comptime Prefabs: type, comptime Components: type, comptime S
         ) !void {
             const ComponentType = Components.getType(comp_name);
             const comp_fields = @typeInfo(ComponentType).@"struct".fields;
-            var component: ComponentType = .{};
+            var component: ComponentType = undefined;
 
             // Process each field in the component type
-            // Note: component is initialized with .{} which sets defaults, so we only
-            // need to handle fields that are explicitly provided in comp_data
             inline for (comp_fields) |comp_field| {
                 const field_name = comp_field.name;
 
@@ -445,8 +443,14 @@ pub fn SceneLoader(comptime Prefabs: type, comptime Components: type, comptime S
                         const data_value = @field(comp_data, field_name);
                         @field(component, field_name) = zon.coerceValue(comp_field.type, data_value);
                     }
+                } else if (comp_field.default_value_ptr) |ptr| {
+                    // Field not provided but has a default value
+                    const default_ptr: *const comp_field.type = @ptrCast(@alignCast(ptr));
+                    @field(component, field_name) = default_ptr.*;
+                } else {
+                    // Required field not provided - compile-time error
+                    @compileError("Missing required field '" ++ field_name ++ "' for component '" ++ comp_name ++ "'");
                 }
-                // Fields not in comp_data retain their defaults from .{} initialization
             }
 
             game.getRegistry().add(parent_entity, component);

--- a/src/zon_coercion.zig
+++ b/src/zon_coercion.zig
@@ -51,6 +51,7 @@ pub fn coerceValue(comptime FieldType: type, comptime data_value: anytype) Field
 
 /// Build a struct from comptime anonymous struct data.
 /// Recursively coerces nested fields.
+/// Raises compile error for missing required fields (fields without defaults).
 pub fn buildStruct(comptime StructType: type, comptime data: anytype) StructType {
     const fields = std.meta.fields(StructType);
     var result: StructType = undefined;
@@ -63,7 +64,7 @@ pub fn buildStruct(comptime StructType: type, comptime data: anytype) StructType
             const default_ptr: *const field.type = @ptrCast(@alignCast(ptr));
             @field(result, field.name) = default_ptr.*;
         } else {
-            @field(result, field.name) = std.mem.zeroes(field.type);
+            @compileError("Missing required field '" ++ field.name ++ "' for struct '" ++ @typeName(StructType) ++ "'");
         }
     }
 

--- a/test/nested_prefab_test.zig
+++ b/test/nested_prefab_test.zig
@@ -152,6 +152,16 @@ pub const ZON_COERCION = struct {
         try expect.equal(result.count, 5);
     }
 
+    test "buildStruct works with required fields when all provided" {
+        const TestStruct = struct {
+            required_value: i32, // No default - required
+            optional_value: u32 = 10,
+        };
+        const result = comptime zon.buildStruct(TestStruct, .{ .required_value = 42 });
+        try expect.equal(result.required_value, 42);
+        try expect.equal(result.optional_value, 10);
+    }
+
     test "tupleToSlice converts tuple to slice" {
         const result = comptime zon.tupleToSlice(i32, .{ 1, 2, 3 });
         try expect.equal(result.len, 3);

